src/lib/ngsi/EntityIdVector.cpp

```diff
@@ -120,6 +120,41 @@ void EntityIdVector::present(const std::string& indent)
 
 /* ****************************************************************************
 *
+* EntityIdVector::lookup - find a matching entity in the entity-vector
+*/
+EntityId* EntityIdVector::lookup(const std::string& id, const std::string& type, const std::string& isPattern)
+{
+  //
+  // isPattern:  "false" or "" is the same
+  //
+  std::string isPatternFromParam = isPattern;
+  if (isPatternFromParam == "")
+  {
+    isPatternFromParam = "false";
+  }
+
+  for (unsigned int ix = 0; ix < vec.size(); ++ix)
+  {
+    std::string isPatternFromVec = vec[ix]->isPattern;
+
+    if (isPatternFromVec == "")
+    {
+      isPatternFromVec = "false";
+    }
+
+    if ((vec[ix]->id == id) && (vec[ix]->type == type) && (isPatternFromVec == isPatternFromParam))
```

Comment:
I think is ok the way it works now... not a big lost of smartness with the current approach.

(I think there is a typo in the example... it should be  { [ E.*, E3 ], [ A1, A2 ] } instead of  { [ E.*, E3 ], [ A1, A3 ] } )

NTC